tests/unit/core/server/request/test-class-core-create-and-confirm-intention-request.php

```diff
@@ -0,0 +1,214 @@
+				$new_class->set_currency_code( 'eur' );
+
+				return $new_class;
+			}
+		);
+		$request->apply_filters( 'cac_test_exception_will_throw_if_immutable_parameter_is_changed_when_filter_is_applied' );
+	}
+
+	public function test_create_intent_request_will_be_created() {
+		$amount   = 1;
+		$currency = 'usd';
+		$pm       = 'pm_1';
+		$cs       = 'cus_1';
+		$cvc      = 'cvc';
+
+		$request = new Create_And_Confirm_Intention( $this->mock_api_client, $this->mock_wc_payments_http_client );
+		$request->set_amount( $amount );
+		$request->set_currency_code( $currency );
+		$request->set_payment_method( $pm );
+		$request->set_customer( $cs );
+		$request->set_capture_method( true );
+		$request->setup_future_usage();
+		$request->set_metadata( [ 'order_number' => 1 ] );
+		$request->set_level3( [ 'level3' => 'level3' ] );
+		$request->set_off_session();
+		$request->set_payment_methods( [ 'pm' => '1' ] );
+		$request->set_cvc_confirmation( $cvc );
+		$this->assertInstanceOf( Create_And_Confirm_Intention::class, $request );
+		$params = $request->get_params();
+
+		$this->assertIsArray( $params );
+		$this->assertSame( $amount, $params['amount'] );
+		$this->assertSame( $currency, $params['currency'] );
+		$this->assertSame( $pm, $params['payment_method'] );
+		$this->assertSame( $cs, $params['customer'] );
+		$this->assertSame( 'manual', $params['capture_method'] );
+		$this->assertSame( 'off_session', $params['setup_future_usage'] );
+		$this->assertArrayHasKey( 'description', $params );
+		$this->assertArrayHasKey( 'metadata', $params );
+		$this->assertSame( 1, $params['metadata']['order_number'] );
+		$this->assertArrayHasKey( 'level3', $params );
+		$this->assertSame( 'true', $params['off_session'] );
+		$this->assertArrayHasKey( 'payment_methods_types', $params );
+		$this->assertSame( $cvc, $params['cvc_confirmation'] );
+		$this->assertSame( 'POST', $request->get_method() );
+		$this->assertSame( WC_Payments_API_Client::INTENTIONS_API, $request->get_api() );
+	}
+
+	public function test_woopay_create_intent_request_will_be_created() {
+		$amount   = 1;
+		$currency = 'usd';
+		$pm       = 'pm_1';
+		$cs       = 'cus_1';
+		$cvc      = 'cvc';
+		$request  = new WooPay_Create_And_Confirm_Intention( $this->mock_api_client, $this->mock_wc_payments_http_client );
+		$request->set_amount( 1 );
+		$request->set_currency_code( 'usd' );
+		$request->set_payment_method( 'pm_1' );
+		$request->set_customer( 'cus_1' );
+		$request->set_capture_method( true );
+		$request->setup_future_usage();
+		$request->set_metadata( [ 'order_number' => 1 ] );
+		$request->set_level3( [ 'level3' => 'level3' ] );
+		$request->set_off_session();
+		$request->set_payment_methods( [ 'pm' => '1' ] );
+		$request->set_cvc_confirmation( 'cvc' );
+		$request->set_is_platform_payment_method();
+		$request->set_has_woopay_subscription();
+		$this->assertInstanceOf( WooPay_Create_And_Confirm_Intention::class, $request );
```

Comment:
I guess initially this started in a different way, but right now you are using the `new` keyword above, so this test is redundant.

Reply:
I agree. Before we had on each set function return statement of self and this self could change class at some point (even it will occur in extremely rare cases). I will remove this line.

Reply:
I guess that's 👍 for removing the method chaining :)

However, the line is still there :p